cli/internal/backends/backends.go

```diff
@@ -2,29 +2,43 @@ package backends
 
 import (
 	"errors"
+	"path/filepath"
 	"turbo/internal/api"
 	"turbo/internal/backends/nodejs"
 	"turbo/internal/fs"
 )
 
 var backends = []api.LanguageBackend{
 	nodejs.NodejsYarnBackend,
+	nodejs.NodejsBerryBackend,
 	nodejs.NodejsNpmBackend,
 	nodejs.NodejsPnpmBackend,
 }
 
-func GetBackend() (*api.LanguageBackend, error) {
+func GetBackend(cwd string) (*api.LanguageBackend, error) {
+	return Detect(cwd)
+}
+
+func Detect(cwd string) (*api.LanguageBackend, error) {
+	possibleBackends := make([]api.LanguageBackend, 0)
 	for _, b := range backends {
-		if fs.FileExists(b.Specfile) &&
-			fs.FileExists(b.Lockfile) {
-			return &b, nil
+		if fs.FileExists(filepath.Join(cwd, b.Specfile)) &&
```

Comment:
I was thinking differently. each backend should implement it's own Detect(cwd) function that can be then called within GetBackend

```go
func GetBackend(cwd string) (*api.LanguageBackend, error) {
	 for _, b := range backends {
    hit, _ := b.Detect(cwd); 
    if hit {
     return b, nil
    }
  }
  return nil, errors.new("could not determine backend")
} 

```

Reply:
Oh sorry, I missunderstood you.

Reply:
All good, the idea here is keep the backend-specifics abstracted from this GetBackend() so when/if we add more backends in the future, we only need to add to the `backends` object and implement the interface